### PR TITLE
add parameter of input in model.summary

### PR DESCRIPTION
**Pull request by wangna11BD**
### PR types
Function optimization
### PR changes
APIs
### Describe
Add parameter of input in model.summary
If forward (self, x, y, z), it can use input_shape; if forward(self, x_list) or forward(self， x_dict), the input of dict or list can only be input.
Examples:
        .. code-block:: python

            import paddle
            import paddle.nn as nn

            class LeNet(nn.Layer):
                def __init__(self, num_classes=10):
                    super(LeNet, self).__init__()
                    self.num_classes = num_classes
                    self.features = nn.Sequential(
                        nn.Conv2D(
                            1, 6, 3, stride=1, padding=1),
                        nn.ReLU(),
                        nn.MaxPool2D(2, 2),
                        nn.Conv2D(
                            6, 16, 5, stride=1, padding=0),
                        nn.ReLU(),
                        nn.MaxPool2D(2, 2))

                    if num_classes > 0:
                        self.fc = nn.Sequential(
                            nn.Linear(400, 120),
                            nn.Linear(120, 84),
                            nn.Linear(
                                84, 10))

                def forward(self, inputs):
                    x = self.features(inputs)

                    if self.num_classes > 0:
                        x = paddle.flatten(x, 1)
                        x = self.fc(x)
                    return x

            lenet = LeNet()

            params_info = paddle.summary(lenet, (1, 1, 28, 28))
            print(params_info)

            # list input demo
            class LeNetListInput(LeNet):

                def forward(self, inputs):
                    x = self.features(inputs[0])

                    if self.num_classes > 0:
                        x = paddle.flatten(x, 1)
                        x = self.fc(x + inputs[1])
                    return x
            
            lenet_list_input = LeNetListInput()
            input_data = [paddle.rand([1, 1, 28, 28]), paddle.rand([1, 400])]
            params_info = paddle.summary(lenet_list_input, input=input_data)
            print(params_info)

            # dict input demo
            class LeNetDictInput(LeNet):

                def forward(self, inputs):
                    x = self.features(inputs['x1'])

                    if self.num_classes > 0:
                        x = paddle.flatten(x, 1)
                        x = self.fc(x + inputs['x2'])
                    return x

            lenet_dict_input = LeNetDictInput()
            input_data = {'x1': paddle.rand([1, 1, 28, 28]),
                          'x2': paddle.rand([1, 400])}
            params_info = paddle.summary(lenet_dict_input, input=input_data)
            print(params_info)

### CODE REVIEW

**Review by dingjiaweiww — DISMISSED** [review]

File: python/paddle/hapi/model_summary.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ def summary(net, input_size, dtypes=None):
                     have multiple input, input_size must be a list which contain 
                     every input's shape. Note that input_size only dim of
                     batch_size can be None or -1.
+        input (dict|list|): the input tensor.
```
Comment (LielinJiang) [inline python/paddle/hapi/model_summary.py]: input can be of any type. And need to explain, if given, input_size and dtype will be ignored.

File: python/paddle/hapi/model_summary.py [hunk 2]
```diff
@@ -77,7 +78,8 @@ def forward(self, inputs):
 
             lenet = LeNet()
 
-            params_info = paddle.summary(lenet, (1, 1, 28, 28))
+            input_data = [paddle.ones([1, 1, 28, 28])]
```
Comment (LielinJiang) [inline python/paddle/hapi/model_summary.py]: do not delete the original examples. Add new examples at the back, and add a example of dict and list input.

File: python/paddle/hapi/model_summary.py [hunk 3]
```diff
@@ -163,7 +165,20 @@ def _check_input(input_size):
...
+        _input = input
+    else:
+        if isinstance(input, dict):
+            _input = []
+            for item in input.keys():
+                _input.append(input[item])
+        elif isinstance(input, list):
+            _input = input
+        else:
+            raise Exception('Input must be list or dict.')
```
Comment (LielinJiang) [inline python/paddle/hapi/model_summary.py]: input can be any type as long as the network can run through

File: python/paddle/hapi/model_summary.py (hunk 1 above)
Comment (wangna11BD) [reply python/paddle/hapi/model_summary.py]: done

File: python/paddle/hapi/model_summary.py (hunk 2 above)
Comment (wangna11BD) [reply python/paddle/hapi/model_summary.py]: done

File: python/paddle/hapi/model_summary.py (hunk 3 above)
Comment (wangna11BD) [reply python/paddle/hapi/model_summary.py]: done

**Review by LielinJiang — DISMISSED** [review]

**Review by dingjiaweiww — DISMISSED** [review]

**Review by LielinJiang — DISMISSED** [review]

**Review by dingjiaweiww — DISMISSED** [review]

**Review by LielinJiang — APPROVED** [review]

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by dingjiaweiww — APPROVED** [review]
